src/plugin.js

```diff
@@ -195,7 +199,7 @@ function doZoom(chart, percentZoomX, percentZoomY, focalPoint, whichAxes) {
 	if (zoomOptions.enabled) {
 		storeOriginalOptions(chart);
 		// Do the zoom here
-		var zoomMode = zoomOptions.mode;
+		var zoomMode = zoomOptions.mode === 'function' ? zoomOptions.mode({chart}) : zoomOptions.mode;
```

Comment:
Should it be `typeof zoomOptions.mode === 'function'` instead of `zoomOptions.mode === 'function'`? (same below)